src/commands/tui/widgets/text_input.rs

```diff
@@ -15,12 +16,21 @@ pub enum TextInputResult {
 }
 
 impl TextInput {
-    pub fn new(text: &str, initial: &str, lines: u16) -> Self {
+    pub fn new(text: Option<&str>, initial: &str, lines: u16, changeable: bool) -> Self {
```

Comment:
Design question: 

A `TextInput` that is not `Changeable` isn't really an `Input`.

Should `TextInput` be renamed in `TextPopup`? Or should there be two separated structures, implementing similar traits?

Reply:
Good idea! I made two blocks how to handle keys.
When we see that there are more differences to come-up, we can always split the widgets in future..